supabase/migrations/20251223184411_encrypt-all-sensitive-data.sql

```diff
@@ -0,0 +1,1126 @@
+-- ++++++++++++++++++++++++++++++++++++
+-- Cashu receive quote changes
+-- ++++++++++++++++++++++++++++++++++++
+
+-- Drop constraint and index on quote_id before dropping the column
+alter table wallet.cashu_receive_quotes drop constraint if exists cashu_receive_quotes_quote_id_key;
+drop index if exists wallet.cashu_receive_quotes_quote_id_key;
+
+-- Drop columns that will be encrypted
+alter table wallet.cashu_receive_quotes drop column amount;
+alter table wallet.cashu_receive_quotes drop column quote_id;
+alter table wallet.cashu_receive_quotes drop column payment_request;
+alter table wallet.cashu_receive_quotes drop column description;
+alter table wallet.cashu_receive_quotes drop column minting_fee;
+alter table wallet.cashu_receive_quotes drop column output_amounts;
+
+-- Add encrypted_data column to store all sensitive fields as encrypted JSON
+alter table wallet.cashu_receive_quotes add column encrypted_data text not null;
+alter table wallet.cashu_receive_quotes add column payment_hash text not null;
+
+-- Add quote_id_hash column to allow looking up quotes by their mint quote_id (hashed for privacy)
+alter table wallet.cashu_receive_quotes add column quote_id_hash text not null;
```

Comment:
should this index include account id too? it can happen that different mints have the same quote id right?